app/store/sagas/location.js

```diff
@@ -0,0 +1,60 @@
+import { call, cancelled, take, takeEvery, race } from "redux-saga/effects";
+import { eventChannel } from "redux-saga";
+
+import { SET_CURRENT_USER_ID, LOGOUT } from "../constants/actionTypes";
+
+import {
+  authenticate,
+  disconnect,
+  watchData,
+  unwatchData
+} from "../services/database";
+
+function subscribe() {
+  return eventChannel(emit => {
+    const handler = snapshot => {
+      emit(snapshot.val());
+    };
+
+    // TODO: promise, call, generators?
+    authenticate();
+    watchData(handler);
+
+    // The subscriber must return an unsubscribe function
+    return () => {
+      unwatchData(handler);
+      disconnect();
+    };
+  });
+}
+
+function* watchDatabase() {
+  const channel = yield call(subscribe);
+
+  try {
+    // eslint-disable no-constant-condition
+    let userSignedOut;
+    while (!userSignedOut) {
```

Comment:
https://firebase.google.com/docs/database/web/offline-capabilities
we could enable offline capabilities!